### PR TITLE
Add `kwarg` to disable manifold checks in constructor

### LINKED ISSUE #536: Add constructor option to Manifold for turning off all checks

### PR DESCRIPTION
**Pull request by luisenp**
Closes #536

### CODE REVIEW

File: theseus/geometry/so3.py [hunk 1]
```diff
@@ -166,12 +176,14 @@ def _log_map_impl(
         return SO3_base.log(self.tensor, jacobians=jacobians)
 
     def _compose_impl(self, so3_2: LieGroup) -> "SO3":
-        return SO3(tensor=SO3_base.compose(self.tensor, so3_2.tensor))
+        return SO3(
+            tensor=SO3_base.compose(self.tensor, so3_2.tensor), disable_checks=True
```
Comment (fantaosha) [inline theseus/geometry/so3.py:180]: Wondering if we should disable checks for `compose()` since composition is one of the major sources for the accumulation of numerical errors.

File: theseus/geometry/so2.py [hunk 2]
```diff
@@ -220,11 +227,11 @@ def _compose_impl(self, so2_2: LieGroup) -> "SO2":
         cos_2, sin_2 = so2_2.to_cos_sin()
         new_cos = cos_1 * cos_2 - sin_1 * sin_2
         new_sin = sin_1 * cos_2 + cos_1 * sin_2
-        return SO2(tensor=torch.stack([new_cos, new_sin], dim=1), strict=False)
+        return SO2(tensor=torch.stack([new_cos, new_sin], dim=1), strict_checks=False)
 
     def _inverse_impl(self, get_jacobian: bool = False) -> "SO2":
         cosine, sine = self.to_cos_sin()
-        return SO2(tensor=torch.stack([cosine, -sine], dim=1), strict=False)
+        return SO2(tensor=torch.stack([cosine, -sine], dim=1), strict_checks=False)
```
Comment (fantaosha) [inline theseus/geometry/so2.py:234]: It is safe for `disable_checks=True`.

File: theseus/geometry/so2.py [hunk 3]
```diff
@@ -220,11 +227,11 @@ def _compose_impl(self, so2_2: LieGroup) -> "SO2":
         cos_2, sin_2 = so2_2.to_cos_sin()
         new_cos = cos_1 * cos_2 - sin_1 * sin_2
         new_sin = sin_1 * cos_2 + cos_1 * sin_2
-        return SO2(tensor=torch.stack([new_cos, new_sin], dim=1), strict=False)
+        return SO2(tensor=torch.stack([new_cos, new_sin], dim=1), strict_checks=False)
```
Comment (fantaosha) [inline theseus/geometry/so2.py:230]: Maybe `disable_checks=True`?

File: theseus/geometry/se3.py [hunk 4]
```diff
@@ -171,10 +181,12 @@ def _log_map_impl(
         return SE3_base.log(self.tensor, jacobians=jacobians)
 
     def _compose_impl(self, se3_2: LieGroup) -> "SE3":
-        return SE3(tensor=SE3_base.compose(self.tensor, se3_2.tensor))
+        return SE3(
+            tensor=SE3_base.compose(self.tensor, se3_2.tensor), disable_checks=True
```
Comment (fantaosha) [inline theseus/geometry/se3.py:185]: Maybe `disable_checks=False` is better since composition (matrix multiplication) might accumulate numerical errors.

File: theseus/geometry/se3.py [hunk 5]
```diff
@@ -157,7 +164,10 @@ def exp_map(
...
         if tangent_vector.ndim != 2 or tangent_vector.shape[1] != 6:
             raise ValueError("Tangent vectors of SE3 can only be 6D vectors.")
-        return SE3(tensor=SE3_base.exp(tangent_vector, jacobians=jacobians))
+        return SE3(
+            tensor=SE3_base.exp(tangent_vector, jacobians=jacobians),
+            disable_checks=True,
+        )
 
     @staticmethod
     def normalize(tensor: torch.Tensor) -> torch.Tensor:
```
Comment (fantaosha) [inline theseus/geometry/se3.py:173]: Should `normalize()` return a group instead of a tensor?

File: theseus/geometry/se2.py [hunk 6]
```diff
@@ -316,7 +323,7 @@ def _compose_impl(self, se2_2: LieGroup) -> "SE2":
         )
         return SE2(
             tensor=torch.cat([new_translation.tensor, new_rotation.tensor], dim=1),
-            strict=False,
+            strict_checks=False,
```
Comment (fantaosha) [inline theseus/geometry/se2.py:326]: `disable_checks=True`.

**Review by fantaosha — CHANGES_REQUESTED** [review]
Only minor changes are needed.

File: theseus/geometry/so2.py (hunk 2 above)
Comment (luisenp) [reply theseus/geometry/so2.py]: Changed.

File: theseus/geometry/so2.py (hunk 3 above)
Comment (luisenp) [reply theseus/geometry/so2.py]: Changed.

File: theseus/geometry/se3.py (hunk 4 above)
Comment (luisenp) [reply theseus/geometry/se3.py]: Changed by `strict_checks=False` so they are normalized if the check fails.

File: theseus/geometry/se2.py (hunk 6 above)
Comment (luisenp) [reply theseus/geometry/se2.py]: Changed.

**Review by fantaosha — APPROVED** [review]
LGTM. Feel free to merge.
